main.go

```diff
@@ -387,14 +391,22 @@ func checkService(opts *Opts, client *k8s.Clientset) error {
 	}
 
 	logger.Infow("Target identified", "IP", serviceIP, "Port", servicePort)
-	go runForwarder(serviceIP, int(servicePort), opts)
+	go runForwarder(serviceIP, servicePort, opts)
 	targetService = service
 	return nil
 }
 
-func runForwarder(serviceIP string, servicePort int, opts *Opts) {
+func runForwarder(serviceIP, servicePort string, opts *Opts) {
 	for {
 		logger.Info("Starting forwarder")
+		fluentTargetIP := serviceIP
+
+		if opts.KonnectivityUDSSocket != "" {
+			ctx, cancel := context.WithCancel(context.Background())
+			go konnectivityproxy.MakeProxy(ctx, opts.KonnectivityUDSSocket, serviceIP, servicePort, logger)
```

Comment:
If opening the proxy fails with `Could not open port for listening`, does the controller shut down or will it just continue to run without being able to forward messages?

Reply:
You were right; in this case the controller would get stuck.

I rewrote the proxy module with (I hope) a cleaner interface that would handle this case too, in a new branch. PR: https://github.com/metal-stack/audit-forwarder/pull/4

(I also added a reference to the source of the proxying code there.)